core/container_exec.go

```diff
@@ -101,7 +101,6 @@ func (container *Container) WithExec(ctx context.Context, opts ContainerExecOpts
 		if err != nil {
 			return nil, fmt.Errorf("register caller: %w", err)
 		}
-		execMD.OTELEnvs = callerOpts.OTELEnvs
```

Comment:
Nothing was actually setting/using `callerOpts.OTELEnvs` anymore so I removed it - but we should bring back its original purpose - see #7510. The regression is already live so this can be fixed in a followup.

Reply:
Should we also remove the `execMD.OTELEnvs` field now?

Reply:
#7510 is fixed in https://github.com/dagger/dagger/pull/7452/commits/8e0e9469a833be635857ea3fd1409fe68e64dbd5 now